### ISSUE TITLE
serverless deploy function ignores package artifact

### ISSUE THREAD

**Issue by ubaniabalogun** [serverless/serverless#3580]
# This is a Bug Report

### What went wrong?
Consider the below serverless.yml file
```
service: test-service

provider:
  name: aws
  runtime: python2.7

functions:
  hello:
    handler: handler.hello
    package:
      artifact: upload.zip
```

When `serverless deploy function -f hello` is run, Serverless ignores the `functions.hello.package.artifact` configuration and falls back to whatever exclude/include configuration it determines.
This indicates that `serverless deploy function -f` is currently ignoring the `functions.[function name].package.artifact` config  
### What did you expect should have happened?
I expected `serverless deploy function -f` to respect the `functions.[function name].package.artifact` configuration the same way `serverless deploy` does.
 
### What was the config you used?
See description

### What stacktrace or error message from your provider did you see?
No error but the behavior is inconsistent with expectations

## Additional Data

* ***Serverless Framework Version you're using***: 1.12.1
* ***Operating System***: OS X El Capitan
* ***Stack Trace***:
* ***Provider Error messages***:

**Comment by eahefnawy** [comment]
@ubaniabalogun `package.artifact` is a property for `sls deploy` only, mainly because you can generate a package with `sls package`, but since there's no `sls package function` command, `sls deploy function` is an independent command that packages and deploys the function.

I understand the confusion though 😊 , I think we better add a note to the docs about this, or maybe add that `sls package function` command? do you feel a need for it? would love your feedback on that one! 🙌

**Comment by ubaniabalogun** [comment]
@eahefnawy Thanks for the response. Let me give you more insight into the problem I'm trying to solve. Perhaps there's already a solution for this in Serverless.

I currently have my own build process (written as a serverless plugin) that creates my function artifacts when `sls deploy` is run. It works beautifully when my intention is to rebuild/redeploy all my functions.

However, I'd like to avoid rebuilding & redeploying all of my functions after making a slight change to just one.

I imagined `sls deploy function` would provide the opportunity to trigger my build process for just one function, but docs have made it clear that `sls deploy function` is better suited for development.

What options (or hooks) are available in the platform to trigger deployment for a single function? I'm working on a project that, at some point in time, will have north of 100 lambda functions in it, so cutting down deployment time is important.

**Comment by eahefnawy** [comment]
@ubaniabalogun if you're doing your own custom packaging with a plugin, I think you need to update it to package individual functions too? In that case you'd be hooking into the `sls deploy function` hooks. I don't think `sls package function` command would help you in that case because you're doing your own custom packaging anyway and not even using `sls package` command.

All the best! 🙌

**Comment by ubaniabalogun** [comment]
@eahefnawy 
> if you're doing your own custom packaging with a plugin, I think you need to update it to package individual functions too? In that case you'd be hooking into the sls deploy function hooks. 

This is exactly how this bug was detected and why I created this bug report.

When I tried to hook into `sls deploy function` hooks, I discovered `sls deploy function` ignores the value in `functions.[function name].package.artifact`, and obeys only exclude/include rules. This makes it impossible to use a build process with the current implementation of `sls deploy function`

I'll update the issue report to reflect that `functions.[function name].package.artifact` is what is being ignored by `sls deploy function`. The current behavior seems like a notable deviation from the current deployment workflow
